python/test/gcs/test_text_to_bigquery.py

```diff
@@ -0,0 +1,196 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+"""
+
+import mock
+import pyspark
+
+from dataproc_templates.gcs.text_to_bigquery import TextToBigQueryTemplate
+import dataproc_templates.util.template_constants as constants
+
+
+class TestTextToBigQueryTemplate:
+    """
+    Test suite for TextToBigQueryTemplate
+    """
+
+    def test_parse_args(self):
+        """Tests TextToBigQueryTemplate.parse_args()"""
+
+        text_to_bigquery_template = TextToBigQueryTemplate()
+        parsed_args = text_to_bigquery_template.parse_args(
+            ["--text.bigquery.input.compression=bzip2",
+             "--text.bigquery.output.mode=append",
+             "--text.bigquery.input.location=gs://test",
+             "--text.bigquery.output.dataset=dataset",
+             "--text.bigquery.output.table=table",
+             "--text.bigquery.temp.bucket.name=bucket",
+             "--text.bigquery.input.delimiter=|"])
+
+        assert parsed_args["text.bigquery.input.compression"] == "bzip2"
+        assert parsed_args["text.bigquery.output.mode"] == "append"
+        assert parsed_args["text.bigquery.input.location"] == "gs://test"
+        assert parsed_args["text.bigquery.output.dataset"] == "dataset"
+        assert parsed_args["text.bigquery.output.table"] == "table"
+        assert parsed_args["text.bigquery.temp.bucket.name"] == "bucket"
+        assert parsed_args["text.bigquery.input.delimiter"] == "|"
+
+    @mock.patch.object(pyspark.sql, 'SparkSession')
+    def test_run_bzip2(self, mock_spark_session):
+        """Tests TextToBigqueryTemplate runs with | delimiter and bzip2 compression format"""
+
+        text_to_bigquery_template = TextToBigQueryTemplate()
+        mock_parsed_args = text_to_bigquery_template.parse_args(
+            ["--text.bigquery.input.location=gs://test",
+             "--text.bigquery.output.dataset=dataset",
+             "--text.bigquery.output.table=table",
+             "--text.bigquery.input.compression=bzip2",
+             "--text.bigquery.temp.bucket.name=bucket",
+             "--text.bigquery.output.mode=ignore",
+             "--text.bigquery.input.delimiter=|"])
+
+        mock_spark_session.read.option().option().option(
+        ).csv.return_value = mock_spark_session.dataframe.DataFrame
+        text_to_bigquery_template.run(mock_spark_session, mock_parsed_args)
+
+        mock_spark_session.read.option.assert_called_with(
+            constants.TEXT_HEADER, True)
+        mock_spark_session.read.option().option.assert_called_with(
+            constants.INPUT_COMPRESSION, "bzip2")
+        mock_spark_session.read.option().option().option.assert_called_with(
+            constants.INPUT_DELIMITER, "|")
+        mock_spark_session.read.option().option().option(
+        ).csv.assert_called_once_with("gs://test")
```

Comment:
PyUnits are failing for all compression types I think you need to add one more `option` in the read section for Infer_schema

Reply:
Resolved.